### PR TITLE
Stop thread pool on shutdown

### PR DESCRIPTION
**Pull request by huachaohuang**
Shutdown without stopping thread pool may cause potentially
core dump. However, stopping thread pool before shutdown may slow down
the shutdown process, unless we can cancel the running task, which
seems not easy to do.

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @zhangjinpeng1987

**Comment by overvenus** [comment]
Please also stop snap-worker thread pool.
https://github.com/pingcap/tikv/blob/fbf6e28a7418eb2d3c2ac3490276141cccf38115/src/server/snap.rs#L182

Maybe `Runable` should **not** provide a default `shutdown()`

**Comment by BusyJay** [comment]
According to experience, TiKV may wait for serveral hours for `snap-worker` to stop.

**Comment by BusyJay** [comment]
May be `exit(0)` in TiKV server is a clean way to stop without wait for all threads to stop, which is unnecessary and possibly time consuming.

**Comment by siddontang** [comment]
IMO, I prefer closing gracefully. 

> TiKV may wait for serveral hours for snap-worker to stop

@BusyJay  Why this takes too much time? Can we quit ASAP?

**Comment by siddontang** [comment]
PTAL

**Comment by BusyJay** [comment]
It depends. If io is slow, then write can be slow, sometimes it may even stall.

File: src/raftstore/store/worker/region.rs [hunk 1]
```diff
@@ -303,4 +303,10 @@ impl Runnable<Task> for Runner {
             } => self.ctx.handle_destroy(region_id, start_key, end_key),
         }
     }
+
+    fn shutdown(&mut self) {
```
Comment (siddontang) [inline src/raftstore/store/worker/region.rs]: seem we don't use this in this PR.

**Comment by siddontang** [comment]
I think we can consider stopping snap-worker later.

File: src/raftstore/store/worker/region.rs (hunk 1 above)
Comment (huachaohuang) [reply src/raftstore/store/worker/region.rs]: I will be called outside automatically.

**Comment by siddontang** [comment]
PTAL

File: src/raftstore/store/worker/region.rs (hunk 1 above)
Comment (zhangjinpeng1987) [reply src/raftstore/store/worker/region.rs]: This thread may need a long time to finish?

**Comment by siddontang** [comment]
CI failed @huachaohuang

**Comment by huachaohuang** [comment]
Uh, we may need to figure out a better way to stop long time task on shutdown.

**Pull request merged**